docs/discovery/provisioning.md

```diff
@@ -0,0 +1,456 @@
+
+## Provisioning Resources and Schemas
+
+### Creating Access Request Schemas
+
+An Access Request Definition tells Marketplace the information that a consumer needs to provide in order for the agent to provision that consumer the ability to use a discovered API.
+
+The Amplify Agent SDK provides a mechanism to set up access request definition schemas based on configuration parameters required by the connected gateway. The agent can set up schemas which register resources of type AccessRequestDefinition in Amplify Central. When the agent discovers an API and publishes it to Amplify Central, the created resource references one of the AccessRequestDefinition resources and associates it with the published API. This could be a schema per API on the target gateway or based on the authentication type of API.
+
+The agent implementation can use the helper method `agent.NewAPIKeyAccessRequestBuilder().Register()` for an access request definition named *api-key* for API Key APIs that require no additional data for provisioning. It is also possible for the agent to create it's own access request definitions by using `agent.NewAccessRequestBuilder()` adding schema properties, see below, before calling Register.
+
+Below is an example of using the builder to create a single Access Request Definition, named after an API. The example only adds a single property, Access Tier, sets it as a required string with an enumeration of accepted. This could be called as part of the discovery process of an API, or at startup if its a single schema for many APIs.
+
+```go
+import (
+    "github.com/Axway/agent-sdk/pkg/agent"
+    "github.com/Axway/agent-sdk/pkg/apic/provisioning"
+)
+
+func createAccessRequestDefinition(apiName string, tiers []string) {
+  agent.NewAccessRequestBuilder().
+    SetName(apiName).
+    AddProperty(provisioning.NewSchemaPropertyBuilder().
+      SetName("tier").
+      SetLabel("Access Tier").
+      SetDescription("The Access Tier").
+      SetRequired()).
+      IsString().
+        SetEnumValues(tiers).
+        SetSortEnumValues()).
+    Register()
+}
+```
+
+### Creating Credential Request Schemas
+
+A Credential Request Definition tells Marketplace the information that a consumer needs to provide in order for the agent to provision that consumer the credentials required by a discovered API.
+
+The Amplify Agent SDK provides a mechanism to set up credential request definition schemas based on configuration parameters required by the connected gateway. The agent can set up schemas which register resources of type CredentialRequestDefinition in Amplify Central. When the agent discovers an API and publishes it to Amplify Central, the created resource references one or more of the CredentialRequestDefinition resources and associates it/them with the published API. This would be a list of all credential types that a particular API will support.
```

Comment:
i had to read this a couple of times to realize that references is a verb.... maybe something like this?  or maybe its just me when reading it 

...discovers an API and publishes it to Amplify Central, the created resource will then reference one or more of the CredentialRequestDefinition resources and associates it